alf/tensor_specs.py

```diff
@@ -86,6 +86,26 @@ def from_array(cls, array, from_dim=0):
         assert isinstance(array, (np.ndarray, np.number))
         return TensorSpec(array.shape[from_dim:], str(array.dtype))
 
+    def replace(self, shape=None, dtype=None) -> TensorSpec:
```

Comment:
Should add argument descriptions and type annotations

Reply:
Sure, updated the PR.